analyzers/src/SonarAnalyzer.CSharp/Rules/EqualityOnFloatingPoint.cs

```diff
@@ -85,12 +85,21 @@ private static void CheckEquality(SonarSyntaxNodeReportingContext context)
                 && IsFloatingPointNumberType(container)
                 && EqualityOperators.Contains(equalitySymbolName))
             {
-                var messageEqualityPart = GetMessageEqualityPart(equals.IsKind(SyntaxKind.EqualsExpression));
-
-                context.ReportIssue(Diagnostic.Create(rule, equals.OperatorToken.GetLocation(), messageEqualityPart));
+                var messageEqualityPart = MessageEqualityPart(equals.IsKind(SyntaxKind.EqualsExpression));
+                var proposed = ProposedMessage(context, equals.Right) ?? ProposedMessage(context, equals.Left) ?? "a range";
+                context.ReportIssue(Diagnostic.Create(Rule, equals.OperatorToken.GetLocation(), messageEqualityPart, proposed));
             }
         }
 
+        private static string ProposedMessage(SonarSyntaxNodeReportingContext context, ExpressionSyntax expression) =>
+            expression is MemberAccessExpressionSyntax memberAccess
+            && memberAccess.GetName() is var memberName
+            && FloatingPointEpsilonDependantMembers.TryGetValue(memberName, out var proposedMethod)
+            && context.SemanticModel.GetSymbolInfo(memberAccess.Expression).Symbol is ITypeSymbol type
+            && IsFloatingPointNumberType(type)
```

Comment:
Cleanup, simplified name with same meaning
```suggestion
            && IsFloatingPointType(type)
```